File: 2º/PDOO/ProyectoJAVA/src/civitas/S_porCasaHotel.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.ArrayList;

/**
 *
 * @author nikitastetskiy
 */
public class S_porCasaHotel extends Sorpresa{
    
    private int valor;
    
    S_porCasaHotel(int valor, String texto){
        super(texto);
        this.valor = valor;
    }
    
    @Override
    void aplicarAJugador(int actual, ArrayList<Jugador> todos) {
        if (jugadorCorrecto(actual, todos)) {
            informe(actual, todos);
            todos.get(actual).modificarSaldo(valor * todos.get(actual).cantidadCasasHoteles());
        }
    }
    
    @Override
    public String toString() {
       return super.toString();
    }
}

File: 2º/PDOO/ProyectoRUBY/lib/s_por_casa_hotel.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require_relative"jugador.rb"
require_relative"sorpresa.rb"

module Civitas
  class S_porCasaHotel<Sorpresa
    def initialize(valor, texto)
      super(texto)
      @valor=valor
    end
    def aplicarAJugador(actual, todos)
      super.informe(actual, todos)
      todos[actual].modificarSaldo(@valor*todos[actual].cantidadCasaHoteles())
    end
    def to_s
      aux=super
      return aux
    end  
  end
end